src/store/catalog-store.ts

```diff
@@ -24,6 +27,21 @@ export class CatalogStore {
       const data = await catalogService.getProducts();
       runInAction(() => {
         this.products = data.results;
+        this.productList = data.results.map(product => ({
+          name: product.masterData.current.name.en || messages.noName,
+          image:
+            product.masterData.current.masterVariant.images[0]?.url ||
+            messages.placeholderJpg,
+          description:
+            product.masterData.current.description.en || messages.noDescription,
+          price: product.masterData.current.masterVariant.prices[0]?.value
+            ? `${(product.masterData.current.masterVariant.prices[0].value.centAmount / 100).toFixed(2)} ${
+                product.masterData.current.masterVariant.prices[0].value
+                  .currencyCode
+              }`
+            : DEFAULT_PRICE,
+          discountPrice: DISCOUNT_PRICE.toString(),
+        }));
```

Comment:
```suggestion
        this.productList = data.results.map(productListMapper);
```
создай под это отдельный отдельный мапер, закинь в утилки